src/hotspot/share/gc/g1/g1Policy.cpp

```diff
@@ -1235,8 +1250,13 @@ void G1Policy::record_pause(PauseKind kind, double start, double end) {
     case LastYoungGC:
       _concurrent_start_to_mixed.add_pause(end - start);
       break;
-    case ConcurrentStartGC:
-      if (_g1h->gc_cause() != GCCause::_g1_periodic_collection) {
+    case ConcurrentStartMarkGC:
+    case ConcurrentStartUndoGC:
+      // Do not track time-to-mixed time for periodic collections as they are likely
+      // to be not representative to regular operation as the mutators are idle at
+      // that time. Also only track full concurrent mark cycles.
+      if ((_g1h->gc_cause() != GCCause::_g1_periodic_collection) &&
+        (kind != ConcurrentStartUndoGC)) {
         _concurrent_start_to_mixed.record_concurrent_start_end(end);
       }
       break;
```

Comment:
I would add the undo-case after to avoid having to check for it in the if-statement. Like this:
```suggestion
    case ConcurrentStartMarkGC:
      // Do not track time-to-mixed time for periodic collections as they are likely
      // to be not representative to regular operation as the mutators are idle at
      // that time.
      if ((_g1h->gc_cause() != GCCause::_g1_periodic_collection) {
        _concurrent_start_to_mixed.record_concurrent_start_end(end);
      }
    case ConcurrentStartUndoGC:
      // Do not track undone concurrent cycles.
      break;
```

If you feel like it you could add an explicit break for the start-case and assert that the start-to-mixed-tracking is not active in the undo-case. But this would require a `is_active()` getter for the tracker.